kratos/geometries/geometry_shape_function_container.h

```diff
@@ -374,8 +372,8 @@ class GeometryShapeFunctionContainer
             return mShapeFunctionsValues[ThisMethod](IntegrationPointIndex, ShapeFunctionIndex);
         if (DerivativeOrderIndex == 1)
             return mShapeFunctionsLocalGradients[ThisMethod][IntegrationPointIndex](ShapeFunctionIndex, DerivativeOrderRowIndex);
-        if (DerivativeOrderIndex > 1)
-            return mShapeFunctionsDerivatives[ThisMethod][IntegrationPointIndex][DerivativeOrderIndex - 2](ShapeFunctionIndex, DerivativeOrderRowIndex);
+        
+        return mShapeFunctionsDerivatives[ThisMethod][IntegrationPointIndex][DerivativeOrderIndex - 2](ShapeFunctionIndex, DerivativeOrderRowIndex);
```

Comment:
that seems alright

Reply:
Yep is just to have a default value if none of the conditions hold, which should be impossible but the compiler does not know that